11-kerberos-ticket-policy.rst

```diff
@@ -0,0 +1,390 @@
+   [29708] 1583503386.372830: Terminating TCP connection to stream AA.BB.CC.DD:88
+   [29708] 1583503386.372831: Response was from master KDC
+   [29708] 1583503386.372832: AS key determined by preauth: aes256-cts/AE1D
+   [29708] 1583503386.372833: Decrypted AS reply; session key is: aes256-cts/12C3
+   [29708] 1583503386.372834: FAST negotiation: available
+   [29708] 1583503386.372835: Initializing KCM:123456 with default princ admin@IPADEMO.LOCAL
+   [29708] 1583503386.372836: Storing admin@IPADEMO.LOCAL -> krbtgt/IPADEMO.LOCAL@IPADEMO.LOCAL in KCM:123456
+   [29708] 1583503386.372837: Storing config in KCM:123456 for krbtgt/IPADEMO.LOCAL@IPADEMO.LOCAL: fast_avail: yes
+   [29708] 1583503386.372838: Storing admin@IPADEMO.LOCAL -> krb5_ccache_conf_data/fast_avail/krbtgt\/IPADEMO.LOCAL\@IPADEMO.LOCAL@X-CACHECONF: in KCM:123456
+   [29708] 1583503386.372839: Storing config in KCM:123456 for krbtgt/IPADEMO.LOCAL@IPADEMO.LOCAL: pa_type: 151
+   [29708] 1583503386.372840: Storing admin@IPADEMO.LOCAL -> krb5_ccache_conf_data/pa_type/krbtgt\/IPADEMO.LOCAL\@IPADEMO.LOCAL@X-CACHECONF: in KCM:123456
+
+As can be seen above, pre-authentication type, ``pa_type``, 151 (SPAKE)
+was used in for pre-authentication. A look at the credential cache content will
+show that the renewal age policy applied corresponds to the ``hardened`` variant::
+
+   [admin@client]$ klist -f -d -e
+   Ticket cache: KCM:123456
+   Default principal: admin@IPADEMO.LOCAL
+
+   Valid starting       Expires              Service principal
+   06.03.2020 09.03.06  06.03.2020 17.03.06  krbtgt/IPADEMO.LOCAL@IPADEMO.LOCAL
+        renew until 07.03.2020 09.03.01, Flags: FRIA
+        Etype (skey, tkt): aes256-cts-hmac-sha1-96, aes256-cts-hmac-sha1-96 , AD types: 
+
+As result, ``admin`` user can now use its ticket up to 8 hours and renew it
+until 48 hours since the original ticket was obtained.
+
+Finally, we can reset the default Kerberos ticket policy to the installation
+default::
+
+  [admin@client]$ ipa krbtpolicy-reset
+    Max life: 86400
+    Max renew: 604800
+
+
+Enforcing authentication indicators
+-----------------------------------
+
+Authentication indicators from the ticket granting ticket are copied by the KDC
+into service tickets issued with the help of the TGT presented by a Kerberos
+client. The indicators can be seen by the applications receiving a communication
+encrypted with the service ticket. This allows an application administrator to
+permit restricted access to only those clients who used specific
+pre-authentication mechanisms to obtain their initial ticket granting ticket.
+For example, an application might decide to only allow access to a specialized
+resource to people who used smart-card authentication initially, even if the
+application itself only supports Kerberos authentication.
+
+At the moment, there are no known applications that implement authentication
+indicator-based authorization. Instead, FreeIPA provides a check for an
```

Comment:
Should we link to how to do that?  I'm thinking of http://web.mit.edu/kerberos/krb5-latest/doc/appdev/gssapi.html#gssapi-authind-attr

Reply:
The target audience of the workshop are system administrators, I don't think it makes sense to add it here.